Tribler/Core/Libtorrent/LibtorrentMgr.py

```diff
@@ -275,6 +275,13 @@ def add_torrent(self, torrentdl, atp):
                 if request_handle:
                     ltsession.remove_torrent(request_handle, 0)
 
+            # Check if we added this torrent before
+            known = ltsession.find_torrent(lt.sha1_hash(infohash))
```

Comment:
What is `known` is `None`? Or is it impossible for this to happen?

Reply:
According to the documentation it always returns a handle, but `is_valid()` will return `False` if the torrent could not be found: http://www.rasterbar.com/products/libtorrent/manual.html#find-torrent-get-torrents , https://github.com/arvidn/libtorrent/blob/ae0baff5a399e75b08d1aa2ac27f84b641c6426b/include/libtorrent/session_handle.hpp#L185 . I tried finding the cpp line as well, but the libtorrent source is a bit too cryptic for me to casually interpret.

Reply:
**tl;dr** Yes it is impossible.

Reply:
Okay great! I guess this one can be merged then 👍